File: backend/src/main/java/com/example/springboot/responses/UserResponse.java
package com.example.springboot.responses;

import lombok.Getter;
import lombok.Setter;

/**
 * A DTO representing the user information returned to clients.
 * This excludes sensitive data like passwords and internal IDs.
 */
@Getter
@Setter
public class UserResponse {
    private String firstName;
    private String lastName;
    private String email;
    private String role;
    private String profilePicture;

    /**
     * Constructs a new UserResponse with the given details.
     *
     * @param firstName      The user's first name.
     * @param lastName       The user's last name.
     * @param email          The user's email address.
     * @param role           The user's role (e.g., USER, ADMIN).
     * @param profilePicture URL to the user's profile picture.
     */
    public UserResponse(String firstName, String lastName, String email, String role, String profilePicture) {
        this.firstName = firstName;
        this.lastName = lastName;
        this.email = email;
        this.role = role;
        this.profilePicture = profilePicture;
    }
}

File: frontend/src/types/UserResponse.ts
/**
 * Represents the user details returned from the API response after a successful login or session retrieval.
 * 
 * @interface UserResponse
 * @property {string} firstName - The user's first name.
 * @property {string} lastName - The user's last name.
 * @property {string} email - The user's email address.
 * @property {string} role - The user's role (e.g., "admin", "user", etc.).
 * @property {string} [profilePicture] - The URL or path to the user's profile picture (optional).
 */
export interface UserResponse {
  firstName: string
  lastName: string
  email: string
  role: string
  profilePicture?: string
}